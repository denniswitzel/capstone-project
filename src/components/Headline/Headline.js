import React from 'react'
import styled from 'styled-components'
import { ReactComponent as Arrow } from '../../images/icons/back-arrow.svg'
import { useHistory } from 'react-router-dom'
import PropTypes from 'prop-types'

Headline.propTypes = { 
    headline: PropTypes.string,
}

Headline.defaultProps = {
    headline: 'Category name'
}

export default function Headline({headline}){
    let history = useHistory();

    function handleClick() {
    history.push("/");
    }

    return(
    <Header>
        <ArrowStyled onClick={handleClick}/>
        <HeadlineStyled>{headline}</HeadlineStyled>
    </Header>
    )
}

const Header = styled.section`
    display: flex;
    grid-column: 1/3;
    margin: 20px 0;
    text-align: center;
`

const HeadlineStyled = styled.h2`
    display: inline;
    position: relative;
    font-weight: 300;
    color: #8E9091;
    border-bottom: 1px solid #4BDB80;
    padding-bottom: 5px;
    margin: 0;
    max-width: 300px;
`
const ArrowStyled = styled(Arrow)`
    position: absolute;
    left: 30px;
    margin-top: 10px;
    cursor: pointer;
`