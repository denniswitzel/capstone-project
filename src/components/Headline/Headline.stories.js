import React from 'react';
import Headline from './Headline'

export default {
  title: 'Product Lists/Headline',
  component: Headline,
  argTypes: {
  headline: { control: 'text'},
  },
};

const Template = (args) => <Headline {...args} />;


export const longHeadline = Template.bind({});
longHeadline.args = {
  headline: 'This is a very long headline to see if it fits',
}

export const shortHeadline = Template.bind({});
shortHeadline.args = {
  headline: 'Headline',
}