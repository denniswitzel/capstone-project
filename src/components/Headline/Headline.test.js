import React from 'react'
import renderer from 'react-test-renderer'
import Headline from './Headline'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

describe('Headline', () => {
  const headline = 'Burgers'

  it('display the headline and an example event', () => {
    const { getByText } = render(<Headline headline={headline} />)
    expect(getByText(headline)).toBeInTheDocument()
  })

  it('renders correctly', () => {
    const tree = renderer.create(<Headline />)
    expect(tree).toMatchSnapshot()
  })
})
