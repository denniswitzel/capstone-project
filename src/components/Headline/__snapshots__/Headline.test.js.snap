// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline renders correctly 1`] = `
<section
  className="sc-AxjAm kQvphW"
>
  <svg
    className="sc-AxiKw jybRSk"
    onClick={[Function]}
  >
    back-arrow.svg
  </svg>
  <h2
    className="sc-AxirZ gMnSyX"
  >
    Category name
  </h2>
</section>
`;
